### ISSUE TITLE
[Time field] VoiceOver does not read spinbuttons correctly in Safari

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3600]
### 😯 Current Behavior

When user place focus on first spinbutton VO reads correctly "Empty, Hour, stepper" but when user clicks arrow right to go to minute stepper VO announces "stepper :" instead of "Empty, Minute, stepper". From there when user click arrow left VO reads "stepper H" instead of "Empty, Hour, stepper".

When user fill hour and minute and use arrows to set focus on each value only the first number from hour is read and minute's value is not announced at all.

https://github.com/user-attachments/assets/cbda2c99-6865-43d2-9a54-03e06c19bbc1

User still can use `control + option + arrow left/right` to get this values announced properly by VO. But it should also work without `control + option` as it works on other browsers.

### 🤔 Expected Behavior/possible solution?

This should work as it works on Chrome:

https://github.com/user-attachments/assets/448035f3-8de2-4b38-89af-68f5a2259ec4

### 🖥️ Steps to Reproduce

1. In Safari go to: https://storybook.sanomalearning.design/?path=/story/form-time-field--basic
2. With VoiceOver turned on place focus on hour stepper
3. Press arrow right 
4. Verify VO announcement
5. Press arrow left
6. Verify VO announcement

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-time-field--basic

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Safari

### Component version

0.1.3

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
